cw_core/lib/crypto_currency.dart

```diff
@@ -269,7 +269,19 @@ class CryptoCurrency extends EnumerableItem<int> with Serializable<int> implemen
       throw  ArgumentError.value(name, 'name', s);
     }
 
-    return CryptoCurrency._nameCurrencyMap[name.toLowerCase()]!;
+    return CryptoCurrency._nameCurrencyMap.values.firstWhere(
+      (element) {
+        final elementTitle = element.title.toLowerCase();
+        final walletTitle = walletCurrency?.title.toLowerCase();
+        final nameLower = name.toLowerCase();
+
+        if (walletCurrency != null) {
+          return elementTitle == nameLower && (element.tag != null ? element.tag?.toLowerCase() == walletTitle : elementTitle == walletTitle);
+        } else {
+          return elementTitle == nameLower;
+        }
+      },
+    );
```

Comment:
no need for all of this, it exactly duplicates the first `try/catch` the issue can be solved by just adding `.toLowerCase()` to `name` on line `256`
```Dart
element.title.toLowerCase() == name.toLowerCase() &&
```